internal/brokers/manager.go

```diff
@@ -160,55 +160,52 @@ func (m *Manager) BrokerFromSessionID(id string) (broker *Broker, err error) {
 
 	broker, exists := m.transactionsToBroker[id]
 	if !exists {
-		return nil, fmt.Errorf("no broker found for session %q", id)
+		return nil, fmt.Errorf("no broker found for session ID %q", id)
 	}
 
 	return broker, nil
 }
 
-// NewSession create a new session for the broker and store the sesssionID on the manager.
-func (m *Manager) NewSession(brokerID, username, lang, mode string) (sessionID string, encryptionKey string, err error) {
+// NewSession create a new session for the broker and store the sessionID on the manager.
+func (m *Manager) NewSession(ctx context.Context, brokerID, username, lang, mode string) (sessionID string, encryptionKey string, err error) {
 	broker, err := m.brokerFromID(brokerID)
 	if err != nil {
 		return "", "", fmt.Errorf("invalid broker: %v", err)
 	}
 
-	sessionID, encryptionKey, err = broker.newSession(context.Background(), username, lang, mode)
+	sessionID, encryptionKey, err = broker.newSession(ctx, username, lang, mode)
 	if err != nil {
 		return "", "", err
 	}
 
 	m.transactionsToBrokerMu.Lock()
 	defer m.transactionsToBrokerMu.Unlock()
-	log.Debugf(context.Background(), "%s: New %s session for %q",
-		sessionID, mode, username)
+	log.Debugf(ctx, "%s: New %s session for %q", sessionID, mode, username)
 	m.transactionsToBroker[sessionID] = broker
 	return sessionID, encryptionKey, nil
 }
 
 // EndSession signals the end of the session to the broker associated with the sessionID and then removes the
 // session -> broker mapping.
-func (m *Manager) EndSession(sessionID string) error {
-	b, err := m.BrokerFromSessionID(sessionID)
-	if err != nil {
-		return err
+func (m *Manager) EndSession(ctx context.Context, sessionID string) error {
+	m.transactionsToBrokerMu.Lock()
+	broker, exists := m.transactionsToBroker[sessionID]
+	if !exists {
+		m.transactionsToBrokerMu.Unlock()
+		return fmt.Errorf("no broker found for session ID %q", sessionID)
 	}
+	delete(m.transactionsToBroker, sessionID)
+	m.transactionsToBrokerMu.Unlock()
 
-	if err = b.endSession(context.Background(), sessionID); err != nil {
+	if err := broker.endSession(ctx, sessionID); err != nil {
 		return err
 	}
 
-	m.transactionsToBrokerMu.Lock()
-	log.Debugf(context.Background(), "%s: End session %q",
-		sessionID, m.transactionsToBroker[sessionID].Name)
-	delete(m.transactionsToBroker, sessionID)
-	m.transactionsToBrokerMu.Unlock()
+	log.Debugf(ctx, "%s: Ended session", sessionID)
 	return nil
 }
 
-// BrokerExists returns true if the brokerID is known by the manager. It can
-// happen that a broker which was stored in the database is not available anymore
-// because the user removed the configuration file.
```

Comment:
I see that it's hard to make sense of this comment. We don't want to remove it without replacement though. I created https://github.com/ubuntu/authd/pull/1084 to handle that.